test/exception-handling.wast

```diff
@@ -69,13 +71,13 @@
         (drop
           (block $l0 (result i32)
             (rethrow
-              (br_on_exn $l0 $e0 (ref.null))
+              (br_on_exn $l0 $e0 (ref.null exn)) ;; was nullref
             )
           )
         )
       )
     )
 
-    (throw $e1 (ref.null))
+    (throw $e1 (ref.null exn)) ;; was nullref
```

Comment:
Didn't know what to do with this test, which is largely invalidated due to the removal of `nullref`, with no more remaining subtypes of `exnref` except `(ref exn)`, which this PR doesn't cover yet.